main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala

```diff
@@ -38,26 +38,29 @@ object EffUnification3 {
   var EnableSmartSubeffecting: Boolean = true
 
   /**
-    * Computes an MGU for **ALL* equations in `eqns`.
+    * Computes an MGU for **ALL* equations in `eqns0`.
     *
-    * Returns `Result.Ok(s)` if *ALL* equations in `eqns` where solvable.
-    * The returned substitution `s` is an MGU for `eqns`.
+    * Returns `Result.Ok(s)` if *ALL* equations in `eqns0` where solvable.
+    * The returned substitution `s` is an MGU for `eqns0`.
     *
-    * Returns `Result.Err(l)` if *a single equation* in `eqns` is unsolvable.
+    * Returns `Result.Err(l)` if *a single equation* in `eqns0` is unsolvable.
     * The returned list `l` is a non-empty list of equations that were unsolvable (i.e. in conflict).
-    * The equations in `l` are derived from `eqns` but are not a strict subset of `eqns`.
-    * That is, an equation in `l` may not directly correspond to any equation in `eqns`. However, their source locations are valid.
+    * The equations in `l` are derived from `eqns0` but are not a strict subset of `eqns0`.
+    * That is, an equation in `l` may not directly correspond to any equation in `eqns0`. However, their source locations are valid.
     *
-    * Returns `Result.Err(eqns)` if `eqns` contains an equation that is ill-kinded. Hence, it is better to handle ill-kinded equations elsewhere.
+    * Returns `Result.Err(eqns0)` if `eqns0` contains an equation that is ill-kinded. Hence, it is better to handle ill-kinded equations elsewhere.
     *
     * Note: Treats `Type.Error` as a constant, i.e. only equal to itself. Hence, it is better to drop equations that contain `Type.Error`.
     */
-  def unifyAll(eqs: List[TypeConstraint.Equality], scope: Scope, renv: RigidityEnv)(implicit flix: Flix): Result[Substitution, List[TypeConstraint]] = {
+  def unifyAll(eqs0: List[TypeConstraint.Equality], scope: Scope, renv: RigidityEnv)(implicit flix: Flix): Result[Substitution, List[TypeConstraint]] = {
     // Performance: Nothing to do if the equation list is empty
-    if (eqs.isEmpty) {
+    if (eqs0.isEmpty) {
       return Result.Ok(Substitution.empty)
     }
 
+    // Randomness: If chaos monkey flag is passed
```

Comment:
Since the flag is not check here, I would change the comment to:

// Randomly reorder the constraints using the chaos monkey.